ecleankernel/layout/blspec.py

```diff
@@ -149,6 +146,23 @@ def find_kernels(self,
         for mkv, fobjs in module_dict.items():
             if any(mkv == k.real_kv for k in kernels.values()):
                 continue
-            kernels.setdefault(mkv, Kernel(mkv)).all_files.extend(fobjs)
+            # this mkv does not have a corresponding kernel
+            # with same real_kv in kernels
+            for layout in ("bls", "uki"):
+                try:
+                    # check if we already have an entry for this mkv
+                    k = kernels[(mkv, layout)]
+                except KeyError:
+                    continue
+                else:
+                    # extend existing entry
+                    k.all_files.extend(fobjs)
+                    break
```

Comment:
Am I missing something or will this only append to one of the kernels when the same version is used in both uki and bls layouts?

Reply:
No, we try to make `k` for both layouts, if both fail we go to the `else` below the `for` loop. If one of them succeeds we go to the `else` below the `try` and append the files. If a version is installed in both layouts the files are appended to the `bls` entry because it is first in the list.

Reply:
If there are two kernel with matching ver, you append to the first one and `break`. The moduledir is never added to the second matching layout.

Reply:
Yes this was intentional, but in hindsight it doesn't really make sense. Fixed!